### DISCUSSION TITLE
Manually set the encoding value after get the text value will not working

### DISCUSSION THREAD

**Discussion by 821938089** [encode/httpx#2041] [Potential Issue]
Since charset_normalizer has the possibility to give wrong encoding (like this), I wrote some code to check the encoding manually, but I found that this doesn't work.
After my research, I found that as long as I get the text value and then set the encoding, then the text value will not change.
I think this is a bug.

```python
import httpx

headers = {'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36',
           'Content-Type': 'application/x-www-form-urlencoded'}

postData = 'searchtype=articlename&keyword=%CE%D2%B5%C4'

url = 'http://www.xntk.org/search.php'

r = httpx.post(url, headers=headers, data=postData)

print(r.encoding) # print 'mac_greek' or 'iso8859_5', charset_normalizer guessed a wrong encoding

if r.text.find('charset=gbk') != -1 or r.text.find('charset=gb2312') != -1 or r.text.find('charset="gbk"') != -1:
    r.encoding = "gbk"
else:
    r.encoding = "utf-8"

print(r.encoding) # gbk
print(r.text) # still print wrong encoding text
```

**Comment by lovelydinosaur** [comment]
This is the behaviour I would expect to see.

I suppose we could do slightly better here if we raised an exception if a user attempts to set the encoding, after having already accessed `r.text`.

**Reply by 821938089** [reply]
This behavior has the potential to cause confusion for the user.
I think it is necessary to throw an exception and instruct the user to use `r.content` to implement this requirement.
And this behavior is inconsistent with `requests`, which should also be detailed in the corresponding place in the documentation.

**Reply by xzmeng** [reply]
I found it counter-intuitive too, especially when you're playing around in an interactive environment, such like Pycharm Jupyter, it has a variable explorer which reads the `text` property when you expand the `Response` object, and then you overwrite the `encoding`, just find nothing changes, feeling confused and look into the source to learn what really happened under the hood.
I suggest it should at least give a warning, if raising an exception is too strict. Or why not just keep consistent with `requests`?
